src/gitlab.ts

```diff
@@ -137,16 +137,77 @@ export const runCommandInGitlabPipeline = async (ctx: Context, task: Task) => {
   */
   await cmdRunner.run("git", ["push", "--force", gitlabRemote, "HEAD"])
 
+  /*
+    Wait until the branch is actually present on GitLab after pushing it. We've
+    noted this measure is required in
+    https://github.com/paritytech/polkadot/pull/5524#issuecomment-1128029579
+    because the pipeline creation request was sent too soon, before GitLab
+    registered the branch, therefore causing the "Reference not found" message.
+  */
+  let wasBranchRegistered = false
+  const waitForBranchMaxTries = 3
+  const waitForBranchRetryDelay = 1024
+
+  const branchPresenceUrl = `https://${
+    gitlab.domain
+  }/api/v4/projects/${encodeURIComponent(
+    gitlabProjectPath,
+  )}/repository/branches/${encodeURIComponent(branchName)}`
+  for (
+    let waitForBranchTryCount = 0;
+    waitForBranchTryCount < waitForBranchMaxTries;
+    waitForBranchTryCount++
+  ) {
+    logger.info(
+      branchPresenceUrl,
+      `Sending request to see if the branch for task ${task.id} is ready`,
+    )
+    const response = await fetch(branchPresenceUrl, {
+      headers: { "PRIVATE-TOKEN": gitlab.accessToken },
```

Comment:
this `PRIVATE-TOKEN` repeats and looks like a good case to either create a const with this header name, or move whole headers with tokens to its own var